### PR TITLE
fix: page scroll with URL hash redirection.

### LINKED ISSUE #4849: Public Event Pages: Organizer Menu Item does not go to "Organizer Section"
The organizer menu item should specifically link to "Organized by.." section on the public event page. Currently it directs to "#owner", e.g. https://eventyay.com/e/16fa59c7#owner, but this does not scroll the page to the "organizer section". It should rather use the tag "#organizer" instead of "#owner" anyways.

Screenshot from 2020-08-23 20-55-46

### PR DESCRIPTION
**Pull request by shantanumallik**
Fixes #4849 

#### Short description of what this resolves:

- Fixed the redirection based on URL hash and anchor tags. Now the anchored div relative to the URL hash shows up at the top of the screen on page load.
- Earlier it required two clicks initially on page load to jump to a particular section of the event page using the side-menu. This was so because the first click was required to bind the anchor scroll action with the click event. It did not trigger the scroll action on the first click. Added scroll action to the first click.

#### Demo Links:
- https://open-event-frontend-d12ttsngk.vercel.app/e/41bb1094#info
- https://open-event-frontend-d12ttsngk.vercel.app/e/41bb1094#tickets
- https://open-event-frontend-d12ttsngk.vercel.app/e/41bb1094#organizer
- https://open-event-frontend-d12ttsngk.vercel.app/e/41bb1094#getting-here

These links direct to the respective sections of the event page in the demo.

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The acceptance, integration, unit tests and linter pass locally with my changes

### CODE REVIEW

**Comment by niranjan94** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6077895&bid=19898004)

File: app/components/public/side-menu.js [hunk 1]
```diff
@@ -13,12 +14,20 @@ export default class SideMenu extends Component {
...
+    }, 20);
+  }
+
   async didInsertElement() {
     super.didInsertElement(...arguments);
     const speakersCall = await this.event.speakersCall;
     this.set('shouldShowCallforSpeakers',
       speakersCall && speakersCall.announcement && (speakersCall.privacy === 'public'));
-
+
```
Comment (niranjan94) [inline app/components/public/side-menu.js:30]: Codacy found an issue: [Trailing spaces not allowed.](https://app.codacy.com/manual/fossasia/open-event-frontend/pullRequest?prid=6077895)

**Comment by shantanumallik** [comment]
@iamareebjamal please review.

**Comment by iamareebjamal** [comment]
Thanks a lot for the fix and explanation. Will review soon and revert

**Comment by iamareebjamal** [comment]
When I click on `Organizer`, the hash should change but it isn't. If you can fix it, please do, otherwise the current behaviour is enough as well

**Comment by shantanumallik** [comment]
@iamareebjamal Now the URL hash updates with the anchor jumps. Please review.

#### Demo:
- https://open-event-frontend-3p2hesww1.vercel.app/e/41bb1094#info
- https://open-event-frontend-3p2hesww1.vercel.app/e/41bb1094#tickets
- https://open-event-frontend-3p2hesww1.vercel.app/e/41bb1094#organizer
- https://open-event-frontend-3p2hesww1.vercel.app/e/41bb1094#getting-here

The links are from the latest vercel deployment.

**Review by iamareebjamal — APPROVED** [review]

File: app/components/public/side-menu.js [hunk 2]
```diff
@@ -13,12 +14,19 @@ export default class SideMenu extends Component {
   @tracked
   showSessions = false;
 
+  async didRender() {
+    super.didRender();
+    const anchor_tag = window.location.hash;
+    $('html, body').animate({
+      scrollTop: $(anchor_tag).offset().top
```
Comment (iamareebjamal) [inline app/components/public/side-menu.js:21]: Throwing `Cannot read property 'top' of undefined`

**Comment by iamareebjamal** [comment]
As mentioned, changes are throwing a lot of errors, hence reverting

File: app/components/public/side-menu.js (hunk 2 above)
Comment (shantanumallik) [reply app/components/public/side-menu.js:21]: Looking into this.

Comment (shantanumallik) [reply app/components/public/side-menu.js:21]: Is this the only error?

Comment (iamareebjamal) [reply app/components/public/side-menu.js:21]: Yes, mostly

**Pull request merged**